packages/core/src/UserState.ts

```diff
@@ -157,7 +157,25 @@ export default class UserState extends Synchronizer {
         return foundLeaf.index
     }
 
-    async getEpochKeys(epoch: bigint) {
+    getEpochKeys(_epoch?: bigint | number, nonce?: number) {
+        const epoch = _epoch ?? this.calcCurrentEpoch()
+        if (
+            typeof nonce === 'number' &&
+            nonce >= this.settings.numEpochKeyNoncePerEpoch
+        ) {
+            throw new Error(
+                `getEpochKeys nonce ${nonce} exceeds max nonce value ${this.settings.numEpochKeyNoncePerEpoch}`
+            )
+        }
+        if (typeof nonce === 'number') {
+            return genEpochKey(
+                this.id.identityNullifier,
+                this.attesterId.toString(),
+                epoch,
+                nonce,
+                this.settings.epochTreeArity ** this.settings.epochTreeDepth
+            )
+        }
```

Comment:
I think there might also be use cases users want to query all time epoch with `nonce=0`
like all airdropped epoch keys in the old unirep social
(or in the future since epoch key nonce can be revealed)

Reply:
So is that a change to this PR or a separate issue?

Reply:
I think it can be changed in this PR?

Reply:
So you mean like return all epoch keys with nonce 0?

Reply:
That seems like something that should be done outside of our library. e.g.

```js
Array(epoch).fill(null).map((_, i) => userState.getEpochKeys(i, 0))
```

Reply:
ah okay
I found that there is no `genEpochKey` in `UserState`
so that make sense to return what `genEpochKey` returns